src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator.ts

```diff
@@ -1,40 +1,49 @@
 import _ from 'lodash/fp'
 import { Fragment, useState } from 'react'
 import { h } from 'react-hyperscript-helpers'
-import { ButtonPrimary, IdContainer } from 'src/components/common'
+import { ButtonPrimary } from 'src/components/common'
 import { centeredSpinner } from 'src/components/icons'
 import Modal from 'src/components/Modal'
 import { Ajax } from 'src/libs/ajax'
-import { reportError } from 'src/libs/error'
+import { withErrorReportingInModal } from 'src/libs/error'
 import Events from 'src/libs/events'
 import { FormLabel } from 'src/libs/forms'
 import * as Nav from 'src/libs/nav'
-import { useCancellation, useOnMount } from 'src/libs/react-utils'
 import * as Utils from 'src/libs/utils'
-import { getExtension, getFileName } from 'src/pages/workspaces/workspace/analysis/file-utils'
-import { analysisNameInput, analysisNameValidator } from 'src/pages/workspaces/workspace/analysis/notebook-utils'
+import { isGoogleWorkspaceInfo, WorkspaceInfo } from 'src/libs/workspace-utils'
+import {
+  AnalysisFile,
+  DisplayName,
+  FileName, getDisplayName,
+  getExtension,
+  getFileName, useAnalysisFiles
+} from 'src/pages/workspaces/workspace/analysis/file-utils'
+import {
+  analysisNameInput,
+  analysisNameValidator
+} from 'src/pages/workspaces/workspace/analysis/notebook-utils'
 import { analysisLauncherTabName } from 'src/pages/workspaces/workspace/analysis/runtime-common'
+import { ToolLabel } from 'src/pages/workspaces/workspace/analysis/tool-utils'
 import validate from 'validate.js'
 
 
-export const AnalysisDuplicator = ({ destroyOld = false, fromLauncher = false, printName, toolLabel, workspaceName, googleProject, workspaceId, namespace, bucketName, onDismiss, onSuccess }) => {
-  const [newName, setNewName] = useState('')
-  const [existingNames, setExistingNames] = useState([])
-  const [nameTouched, setNameTouched] = useState(false)
-  const [processing, setProcessing] = useState(false)
-  const signal = useCancellation()
+export interface AnalysisDuplicatorProps {
+  destroyOld: boolean
```

Comment:
Where there are default values, props should possibly be marked optional.

```suggestion
  destroyOld?: boolean
```